Release

antariuz opened the pull request.

antariuz committed:
working on

antariuz committed:
All is done

antariuz committed:
Done read/write throw IO Streams, FileReader/FileWriter;

antariuz committed:
Done: Charset fixes in FileReader/FileWriter version

iddqd2d commented on src/regExpTraining/Main.java:
```diff
@@ -1,8 +1,142 @@
+        while (matcher.find()) {
+            matchesList.add(matcher.group());
+        }
+        if (matchesList.size() == 0) throw new MyException("Nothing is found");
```
Есть же isEmpty()

iddqd2d commented on src/regExpTraining/Main.java:
```diff
@@ -1,8 +1,142 @@
+        FileReader reader2 = new FileReader(file2, myCharset);
+        BufferedReader bufferedReader1 = new BufferedReader(reader1);
+        BufferedReader bufferedReader2 = new BufferedReader(reader2);
+        String readFile1 = "";
```
Используй StringBuilder

iddqd2d commented on src/regExpTraining/Main.java:
```diff
@@ -1,8 +1,142 @@
+        return matchesList;
+    }
+
+    public static void main(String[] args) throws MyException, IOException {
```
Обработай исключения, а не прикинь их

antariuz committed:
working on fixes to handle Exceptions

antariuz committed:
Done: code has been rewritten

antariuz committed:
Done: package name fix

antariuz committed:
Done:
-rewritten with try-with-resources
-IOStream: added NullPointerException exception section
